woocommerce/woocommerce-gutenberg-products-block
# Cart & Checkout blocks: print all item data and update variation designs

This PR:
* prints the cart item data added by extensions to the Cart and Checkout blocks (fixes #3138):
  * this will be useful to display subscription details like sign up fees, free trials, etc. (see follow-up  #3659)
  * some other extensions also add cart item data, like Product Add-Ons
* updates the styles of variation data in Cart and Checkout blocks so they are aligned with the lastest designs (fixes #3630).

### Screenshots

In the screenshots below:

* Magazine is a subscription product.
* Hoodie is a variable product.
* Cap is a product with add-ons.

_Cart:_
![imatge](https://user-images.githubusercontent.com/3616980/104019093-3c758100-51bb-11eb-931a-9a330fc7d940.png)

_Checkout:_
<img src="https://user-images.githubusercontent.com/3616980/104019138-4d25f700-51bb-11eb-8769-87d8bed590eb.png" alt="Screenshot" width="498"/>

### How to test the changes in this Pull Request:

Variations:

1. Add a variable product to your cart and go to the Cart and Checkout pages.
2. Verify variations are displayed with the new design (a list instead of separated by `/`).

WC Subscriptions:

1. Currently, WC Subscriptions doesn't use cart item data yet, but you can checkout this WIP branch to test it: `fix/subscription-details-cart-item-data`.
2. Create a subscription product with free trial and/or sign up fee.
3. Add that product to your cart and go to the Cart and Checkout pages.
4. Verify the free trial and sign up fee are displayed.

WC Product Add-Ons:

1. Create a product with a product add-on.
2. Add that product to your cart and go to the Cart and Checkout pages.
3. Verify add-on data is displayed.

### Changelog

> Extensibility: show item data in Cart and Checkout blocks and update the variation data styles.
